pkg/api/types.go

```diff
@@ -0,0 +1,151 @@
+}
+
+// Properties represents the ACS cluster definition
+type Properties struct {
+	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
+	OrchestratorProfile     *OrchestratorProfile     `json:"orchestratorProfile,omitempty"`
+	MasterProfile           *MasterProfile           `json:"masterProfile,omitempty"`
+	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty"`
+	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
+	AzProfile               *AzProfile               `json:"azProfile,omitempty"`
+}
+
+// AzProfile holds the azure context for where the cluster resides
+type AzProfile struct {
+	TenantID       string `json:"tenantId,omitempty"`
+	SubscriptionID string `json:"subscriptionId,omitempty"`
+	ResourceGroup  string `json:"resourceGroup,omitempty"`
+}
+
+// ServicePrincipalProfile contains the client and secret used by the cluster for Azure Resource CRUD
+type ServicePrincipalProfile struct {
+	ClientID string `json:"clientId"`
+	Secret   string `json:"secret,omitempty" conform:"redact"`
+}
+
+// ProvisioningState represents the current state of container service resource.
+type ProvisioningState string
+
+const (
+	// Creating means ContainerService resource is being created.
+	Creating ProvisioningState = "Creating"
+	// Updating means an existing ContainerService resource is being updated
+	Updating ProvisioningState = "Updating"
+	// Failed means resource is in failed state
+	Failed ProvisioningState = "Failed"
+	// Succeeded means resource created succeeded during last create/update
+	Succeeded ProvisioningState = "Succeeded"
+	// Deleting means resource is in the process of being deleted
+	Deleting ProvisioningState = "Deleting"
+	// Migrating means resource is being migrated from one subscription or
+	// resource group to another
+	Migrating ProvisioningState = "Migrating"
+	// Upgrading means an existing ContainerService resource is being upgraded
+	Upgrading ProvisioningState = "Upgrading"
+)
+
+// OrchestratorProfile contains Orchestrator properties
+type OrchestratorProfile struct {
+	OrchestratorVersion string           `json:"orchestratorVersion"`
+	OpenShiftConfig     *OpenShiftConfig `json:"openshiftConfig,omitempty"`
+}
+
+// OpenShiftConfig holds configuration for OpenShift
+type OpenShiftConfig struct {
+	PublicHostname string
+	RouterProfiles []OpenShiftRouterProfile
+}
+
+// OpenShiftRouterProfile represents an OpenShift router.
+type OpenShiftRouterProfile struct {
+	Name            string
+	PublicSubdomain string
+	FQDN            string
+}
+
+// MasterProfile represents the definition of the master cluster
+type MasterProfile struct {
+	// Master LB public endpoint/FQDN with port
+	// The format will be FQDN:2376
+	// Not used during PUT, returned as part of GET
+	FQDN string `json:"fqdn,omitempty"`
+}
+
+// Extension represents an extension definition in the master or agentPoolProfile
```

Comment:
This can be removed. I don't think it's using anywhere.

Reply:
:+1: